examples/posenet/src/PoseNet.js

```diff
@@ -0,0 +1,372 @@
+    let heatmap_dimension = [1, (this._inputs[1]-1)/this._outputstride+1, (this._inputs[2]-1)/this._outputstride+1, 17];
+    let heatmap = {type: type, dimensions: heatmap_dimension};
+    let offset_dimension = [1, (this._inputs[1]-1)/this._outputstride+1, (this._inputs[2]-1)/this._outputstride+1, 34];
+    let offset = {type: type, dimensions: offset_dimension};
+    let input = {type:type, dimensions: this._inputs};
+    let tensorId_input = this._operandIndex++;
+    this._model.addOperand(input);
+    let tensorId_output = [];
+    tensorId_output.push(this._operandIndex++);
+    this._model.addOperand(heatmap);
+    tensorId_output.push(this._operandIndex++);
+    this._model.addOperand(offset);
+    if(this._type === "Multiperson"){
+      let displacement_fwd_dimension = [1, (this._inputs[1]-1)/this._outputstride+1, (this._inputs[2]-1)/this._outputstride+1, 32];
+      let displacement_bwd_dimension = [1, (this._inputs[1]-1)/this._outputstride+1, (this._inputs[2]-1)/this._outputstride+1, 32];
+      let displacement_fwd = {type:type, dimensions:displacement_fwd_dimension};
+      let displacement_bwd = {type:type, dimensions:displacement_bwd_dimension};
+      tensorId_output.push(this._operandIndex++);
+      this._model.addOperand(displacement_fwd);
+      tensorId_output.push(this._operandIndex++);
+      this._model.addOperand(displacement_bwd);
+    }
+    this._model.identifyInputsAndOutputs([tensorId_input], tensorId_output);
+  }
+
+  _calculateOutput(input_dimension, shape, stride, layer){
+    var padding = 1;
+    var output_dimension;
+    if(layer === "conv2d"){
+      output_dimension = [1, Math.floor((input_dimension[1]-shape[1]+2)/stride+1), 
+                          Math.floor((input_dimension[2]-shape[2]+2)/stride+1), shape[0]];
+    }
+    else if(layer === "depthwise"){
+      output_dimension = [1, Math.floor((input_dimension[1]-shape[1]+2)/stride+1), 
+                          Math.floor((input_dimension[2]-shape[2]+2)/stride+1), input_dimension[3]];
+    }
+    else{
+      output_dimension = [1, Math.floor((input_dimension[1]-shape[1]+0)/stride+1), 
+                          Math.floor((input_dimension[2]-shape[2]+0)/stride+1), shape[0]];
+    }
+    return output_dimension;
+  }
+
+
+  _addScalarInt32(value){
+    const scalarInt32Type = {type: this._nn.INT32};
+    let index = this._operandIndex++;
+    this._model.addOperand(scalarInt32Type);
+    this._model.setOperandValue(index, new Int32Array([value]));
+    return index;
+  }
+  _addScalarFloat32(value) {
+    const scalarInt32Type = {type: this._nn.FLOAT32};
+    let index = this._operandIndex++;
+    this._model.addOperand(scalarInt32Type);
+    this._model.setOperandValue(index, new Float32Array([value]));
+    return index;
+	}
+  
+  _addOpsAndParams(){
+    let index = 0;
+    for(let i in this._tfmodel){
+      if(this._tfmodel[i].convType=="conv2d"){
+        let inputs = [];
+        inputs.push(this._model._inputs[0]);
```

Comment:
avoid to access internal properties of `this._model`. When running on `WebML`, `this._model` doesn't have `_inputs` which will raise error. Please use other properties to store inputs, e.g. `this._inputTensorId`.